mne/io/tests/test_meas_info.py

```diff
@@ -410,50 +427,86 @@ def test_check_consistency():
+
+        anonymous_file_id_secs = xx.info['file_id']['secs'] == DATE_NONE[0]
+        anonymous_file_id_usecs = xx.info['file_id']['usecs'] == DATE_NONE[1]
+        anonymous_meas_id_secs = xx.info['meas_id']['secs'] == DATE_NONE[0]
+        anonymous_meas_id_usecs = xx.info['meas_id']['usecs'] == DATE_NONE[1]
+        if xx.info['meas_date'] is None:
+            anonymous_meas_date = True
+        else:
+            assert isinstance(xx.info['meas_date'], tuple)
+            anonymous_meas_date = xx.info['meas_date'] == DATE_NONE
+        if not hasattr(xx, 'annotations'):
+            anonymous_annotations = True
+        else:
+            if xx.annotations is None:
+                anonymous_annotations = True
+            else:
+                anonymous_annotations = xx.annotations.orig_time is None
+
+        return anonymity_checks(missing_subject_info,
+                                anonymous_file_id_secs,
+                                anonymous_file_id_usecs,
+                                anonymous_meas_id_secs,
+                                anonymous_meas_id_usecs,
+                                anonymous_meas_date,
+                                anonymous_annotations)
+
     pytest.raises(TypeError, anonymize_info, 'foo')
 
     # Fake some subject data
     raw = read_raw_fif(raw_fname)
+    raw.set_annotations(Annotations(onset=[0, 1],
+                                    duration=[1, 1],
+                                    description='dummy',
+                                    orig_time=None))
     raw.info['subject_info'] = dict(id=1, his_id='foobar', last_name='bar',
                                     first_name='bar', birthday=(1987, 4, 8),
                                     sex=0, hand=1)
 
-    orig_file_id = raw.info['file_id']['secs']
-    orig_meas_id = raw.info['meas_id']['secs']
+    # Test no error for incomplete info
+    info = raw.info.copy()
+    info.pop('file_id')
+    anonymize_info(info)
+
     # Test instance method
     events = read_events(event_name)
     epochs = Epochs(raw, events[:1], 2, 0., 0.1)
-    for inst in [raw, epochs]:
-        assert 'subject_info' in inst.info.keys()
-        assert inst.info['subject_info'] is not None
-        for key in ('file_id', 'meas_id'):
-            assert inst.info[key]['secs'] != DATE_NONE[0]
-            assert inst.info[key]['usecs'] != DATE_NONE[1]
-        assert isinstance(inst.info['meas_date'], tuple)
-        assert inst.info['meas_date'] != DATE_NONE
-        inst.anonymize()
-        assert 'subject_info' not in inst.info.keys()
-        for key in ('file_id', 'meas_id'):
-            assert inst.info[key]['secs'] == DATE_NONE[0]
-            assert inst.info[key]['usecs'] == DATE_NONE[1]
-        assert inst.info['meas_date'] is None
+
+    assert not any(_is_anonymous(raw))
+    raw.anonymize()
+    assert all(_is_anonymous(raw))
+
+    assert not any(_is_anonymous(epochs)[:-1])  # epochs has no annotations
```

Comment:
this `[:-1]` show that `_is_anonymous` should depend on the object. So we either do it an object method or we make a function that duck-types inside? 

```py
assert not any(raw._is_anoymous())
raw.anonymize
assert all(raw._is_anonymous())
```
vs
```py
from mne import is_anonymous

assert not any(is_anoymous(raw))
raw.anonymize
assert all(is_anonymous(raw))

with pytest.raises(RuntimeError, msg="42 is awesome"):
    is_anonymous(42)
```

Reply:
or 
```py
assert is_anonymous(42) is True
```
(and maybe warn something like the parameter has no fields with sensible data)

Reply:
_is_anonymous is meant to be hidden from users. If you make it an object method it will appear in the tab complete maybe in the documentation of the class. I am for hiding the complexity as much as possible.

Reply:
at this point `_is_anonymous` is a helper function that its only inside the test.